components/NewRegistrationForm/NewRegistrationForm.js

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { login } from 'common/utils/auth-utils';
+import { insertIf } from 'common/utils/array-utils';
+import MultiStepForm from 'components/Form/MultiStepForm';
+import {
+  InitialStep,
+  PersonalDetailsStep,
+  ClassificationStep,
+  MilitaryStep,
+  TechnologyStep,
+} from './Steps';
+
+class NewRegistrationForm extends Component {
+  static propTypes = {
+    initialValues: PropTypes.objectOf(PropTypes.string),
+  };
+
+  static defaultProps = {
+    initialValues: {
+      // order doesn't matter here
+      ...InitialStep.initialValues,
+      ...PersonalDetailsStep.initialValues,
+      ...ClassificationStep.initialValues,
+      ...MilitaryStep.initialValues,
+      ...TechnologyStep.initialValues,
+    },
+  };
+
+  state = {
+    shouldShowMilitaryStep: false,
+  };
+
+  requireMilitaryStep = () => {
+    this.setState({ shouldShowMilitaryStep: true });
+  };
+
+  hideMilitaryStep = () => {
+    this.setState({ shouldShowMilitaryStep: false });
+  };
+
+  // This method was only created to modularize logic
+  defineSteps = () => {
```

Comment:
Are these objects really necessary?  I think you could just use the Components themselves since `validationSchema` and `submitHandler` are both static on all of the steps.  Something like:

```
getSteps = () => [
    InitialStep,
    PersonalDetailsStep,
    ClassificationStep,
    // eslint-disable-next-line react/destructuring-assignment
    ...insertIf(this.state.shouldShowMilitaryStep, MilitaryStep),
    TechnologyStep,
  ];

  render() {
    const { initialValues } = this.props;

    return (
      <MultiStepForm
        initialValues={initialValues}
        onFinalStepSuccess={login}
        steps={this.getSteps()}
      />
    );
  }
```

And in MultiStepForm (note last line):
```
  render() {
    const { initialValues, steps } = this.props;
    const { errorMessage, stepNumber } = this.state;

    const CurrentStep = steps[stepNumber];
    const isFirstStep = stepNumber === 0;

    return (
      <Formik
        initialValues={initialValues}
        validationSchema={CurrentStep.validationSchema}
        onSubmit={this.handleSubmit}
        render={formikBag => (
          <Form className={styles.MultiStepForm} onSubmit={formikBag.handleSubmit}>
            {/*{CurrentStep.render(formikBag)}*/}
            <CurrentStep {...formikBag} />
           .....
```